packages/zoe/src/contracts/loan/updateDebt.js

```diff
@@ -30,48 +31,72 @@ export const makeDebtCalculator = debtCalculatorConfig => {
     calcInterestFn = calculateInterest,
     originalDebt,
     loanMath,
-    periodAsyncIterable,
+    periodNotifier,
     interestRate,
+    interestPeriod,
     zcf,
     configMinusGetDebt,
   } = debtCalculatorConfig;
   let debt = originalDebt;
 
+  let lastCalculationUpdate;
+
   const {
     updater: debtNotifierUpdater,
     notifier: debtNotifier,
   } = makeNotifierKit();
 
   const getDebt = () => debt;
 
-  const config = { ...configMinusGetDebt, getDebt };
+  const config = { ...configMinusGetDebt, getDebt, interestPeriod };
 
-  const updateDebt = _state => {
-    const interest = loanMath.make(calcInterestFn(debt.value, interestRate));
-    debt = loanMath.add(debt, interest);
-    debtNotifierUpdater.updateState(debt);
-    scheduleLiquidation(zcf, config);
+  const updateDebt = state => {
+    let prevUpdateTimestamp = lastCalculationUpdate.value;
+    const { value: newTimestamp } = state;
+    while (prevUpdateTimestamp + interestPeriod <= newTimestamp) {
+      prevUpdateTimestamp += interestPeriod;
+      const interest = loanMath.make(calcInterestFn(debt.value, interestRate));
+      debt = loanMath.add(debt, interest);
+      debtNotifierUpdater.updateState(debt);
+    }
   };
 
-  /** @type {IterationObserver<undefined>} */
-  const debtObserver = {
-    // Debt is updated with interest every time the
-    // periodAsyncIterable pushes another value
-    updateState: updateDebt,
-    finish: updateDebt,
-    fail: reason => {
-      debtNotifierUpdater.fail(reason);
-      throw Error(reason);
-    },
-  };
-  harden(debtObserver);
+  function addToDebtWhenNotified(lastCount) {
```

Comment:
Let's not use the `function` notation since that's not used elsewhere in this file.